src/map.cpp

```diff
@@ -155,6 +155,17 @@ VoxelObject::~VoxelObject()
  * @return Sprite to display for the voxel object.
  */
 
+/**
+ * Get the overlay sprite(s) to draw for the voxel object.
+ * @param sprites Sprites at the right size for drawing.
+ * @param orient Direction of view.
+ * @return Sprites to overlay for the voxel object, with their respective recolourings.
+ */
+std::vector<std::pair<const ImageData*, const Recolouring*>> VoxelObject::GetOverlays(const SpriteStorage *sprites, ViewOrientation orient) const
```

Comment:
It's always tempting to stick an available solution like `pair` into there when you are just outside standard types. The cost of that is a long type (which is mostly fixable with a `typedef`, so not much of a problem), and at every spot where you need to deal with this data, non-intuitive `.first` and `.second` field names, and no way to attach some intelligence into the `pair` by adding a method to it, leading to shorter and more readable code.

I am not sure if using a self written class here instead of `pair` improves things, but in my experience the threshold for that is pretty low in most cases. Please consider.